### PR TITLE
feat(endpoint): make gate endpoint configurable

### PR DESCRIPTION
**Pull request by ethanfrogers**
makes the `gateEndpoint` configurable via the config file. the flag
still works if the property is missing.

fixes spinnaker/spinnaker#3094

### CODE REVIEW

**Comment by ethanfrogers** [comment]
cc @jtk54

File: command/meta.go [hunk 1]
```diff
@@ -75,10 +72,10 @@ type ApiMeta struct {
 func (m *ApiMeta) GlobalFlagSet(cmd string) *flag.FlagSet {
 	f := flag.NewFlagSet(cmd, flag.ContinueOnError)
 
-	f.StringVar(&m.gateEndpoint, "gate-endpoint", "http://localhost:8084",
+	f.StringVar(&m.Config.GateEndpoint, "gate-endpoint", "http://localhost:8084",
```
Comment (jtk54) [inline command/meta.go]: Not sure this is evaluated in the right order, I would expect `--gate-endpoint` to override what is specified in the config.

**Comment by ethanfrogers** [comment]
@jtk54 sorry for the delay, last week was crazy busy for me!

**Comment by jtk54** [comment]
Looks like the `--gate-endpoint` flag isn't working properly, [tests](https://github.com/spinnaker/spin/blob/master/command/applications/application_save_test.go#L41) are breaking. I'd consider keeping the top-level `gateEndpoint` and do the merge manually. I think the flag should override the config property if is it supplied.

**Comment by ethanfrogers** [comment]
@jtk54 i know, i need to find some time to fix it. i'm going to go ahead and close this out until i get a chance to come back to it.

**Pull request closed without merging**